### PR TITLE
feat (3rd_party_board): Add new Adafruit Feather C6 - rev B

### PR DESCRIPTION
**Pull request by ladyada**
feat: add feather esp32 c6
tested with hardware!

image

### CODE REVIEW

**Comment by ladyada** [comment]
feat: add feather esp32-c6 new hardware

File: boards.txt [hunk 1]
```diff
@@ -13299,6 +13299,185 @@ adafruit_feather_esp32s3_reversetft.menu.ZigbeeMode.zczr=Zigbee ZCZR (coordinato
...
+adafruit_feather_esp32c6.menu.EraseFlash.none.upload.erase_cmd=
+adafruit_feather_esp32c6.menu.EraseFlash.all=Enabled
+adafruit_feather_esp32c6.menu.EraseFlash.all.upload.erase_cmd=-e
+
+adafruit_feather_esp32c6.menu.ZigbeeMode.default=Disabled
+adafruit_feather_esp32c6.menu.ZigbeeMode.default.build.zigbee_mode=
+adafruit_feather_esp32c6.menu.ZigbeeMode.default.build.zigbee_libs=
+adafruit_feather_esp32c6.menu.ZigbeeMode.ed=Zigbee ED (end device)
+adafruit_feather_esp32c6.menu.ZigbeeMode.ed.build.zigbee_mode=-DZIGBEE_MODE_ED
+adafruit_feather_esp32c6.menu.ZigbeeMode.ed.build.zigbee_libs=-lesp_zb_api_ed -lesp_zb_cli_command -lzboss_stack.ed.trace -lzboss_stack.ed -lzboss_port
```
Comment (P-R-O-C-H-Y) [inline boards.txt:13471]: Please remove the -lzboss_stack.ed.trace from linker, as it's no longer available.
```suggestion
adafruit_feather_esp32c6.menu.ZigbeeMode.ed.build.zigbee_libs=-lesp_zb_api_ed -lesp_zb_cli_command -lzboss_stack.ed -lzboss_port
```

File: boards.txt [hunk 2]
```diff
@@ -13299,6 +13299,185 @@ adafruit_feather_esp32s3_reversetft.menu.ZigbeeMode.zczr=Zigbee ZCZR (coordinato
...
+adafruit_feather_esp32c6.menu.ZigbeeMode.zczr=Zigbee ZCZR (coordinator)
+adafruit_feather_esp32c6.menu.ZigbeeMode.zczr.build.zigbee_mode=-DZIGBEE_MODE_ZCZR
+adafruit_feather_esp32c6.menu.ZigbeeMode.zczr.build.zigbee_libs=-lesp_zb_api_zczr -lesp_zb_cli_command -lzboss_stack.zczr.trace -lzboss_stack.zczr -lzboss_port
```
Comment (P-R-O-C-H-Y) [inline boards.txt:13474]: Please remove the -lzboss_stack.zczr.trace from linker, as it's no longer available.
```suggestion
adafruit_feather_esp32c6.menu.ZigbeeMode.zczr.build.zigbee_libs=-lesp_zb_api_zczr -lesp_zb_cli_command -lzboss_stack.zczr -lzboss_port
```

**Review by P-R-O-C-H-Y — CHANGES_REQUESTED** [review]
@ladyada Looking good, just small fix needed on Zigbee.

**Comment by me-no-dev** [comment]
@P-R-O-C-H-Y PTAL

**Review by P-R-O-C-H-Y — APPROVED** [review]

**Comment by me-no-dev** [comment]
@hathach please apply the changes requested by the pre-commit hook: https://github.com/espressif/arduino-esp32/actions/runs/9748594236/job/26903898166?pr=9961#step:8:28

They could not be automatically applied to PRs from organizations. Thanks!

**Comment by hathach** [comment]
@me-no-dev thanks for reminder, I have fixed the pre-commit clang format

**Pull request merged**